core/domain/exp_services.py

```diff
@@ -1723,6 +1723,34 @@ def revert_version_history(
         new_version_history_model.put()
 
 
+def get_exploration_version_valid_info(exploration_id, revert_to_version):
+    """Tests whether an exploration can be reverted to the given version
+    number. Does not commit any changes.
+
+    Args:
+        exploration_id: str. The id of the exploration to be reverted to the
+            current version.
+        revert_to_version: int. The version to which the given exploration
+            is to be reverted.
+
+    Returns:
+        dict{'valid': bool, 'details': Optional[str]}. If the revert_to_version
+        passes all backend validation checks, then 'details' is None.
+        Otherwise, 'details' stores the validation error as a string.
+    """
+    # Validate the previous version of the exploration.
+    exploration = exp_fetchers.get_exploration_by_id(
+        exploration_id, version=revert_to_version)
+    exploration_rights = rights_manager.get_exploration_rights(exploration.id)
+    try:
+        exploration.validate(
+            exploration_rights.status == rights_domain.ACTIVITY_STATUS_PUBLIC)
+    except Exception as ex:
+        return {'valid': False, 'details': str(ex)}
+
+    return {'valid': True, 'details': None}
```

Comment:
Should we just return str or None from here and then form the rest in the controller? I think in services we mostly don't directly generate the dictionaries to be returned from the controller.

Reply:
Got it -- fixed!